Makefile

```diff
@@ -308,11 +308,11 @@ $(BUILD_DIR)/src/libultra/rmon/%.o: CC := $(CC_OLD)
 $(BUILD_DIR)/src/code/jpegutils.o: CC := $(CC_OLD)
 $(BUILD_DIR)/src/code/jpegdecoder.o: CC := $(CC_OLD)
 
-$(BUILD_DIR)/src/boot/%.o: CC := $(PYTHON) tools/asm_processor/build.py $(CC) -- $(AS) $(ASFLAGS) --
-$(BUILD_DIR)/src/code/%.o: CC := $(PYTHON) tools/asm_processor/build.py $(CC) -- $(AS) $(ASFLAGS) --
-$(BUILD_DIR)/src/overlays/%.o: CC := $(PYTHON) tools/asm_processor/build.py $(CC) -- $(AS) $(ASFLAGS) --
+# For using asm_processor on some files:
+#$(BUILD_DIR)/.../%.o: CC := $(PYTHON) tools/asm_processor/build.py $(CC) -- $(AS) $(ASFLAGS) --
+
+$(BUILD_DIR)/%.o: CC := tools/reencode.sh $(CC)
```

Comment:
good remark

It does break, I think because `reencode.sh` expects the input file to be the last argument,
but the permuter uses `[compile command (including reencode.sh)] input -o output`
instead of I-think-would-work `[compile command (including reencode.sh)] -o output input`

<details>

```
$ ../decomp-permuter/import.py src/code/z_play.c expected/build/gc-eu-mq/functions/src/code/z_play/Play_Update.s VERSION=gc-eu-mq
Compiler type: ido
Function name: Play_Update
Compiler: tools/reencode.sh tools/ido_recomp/linux/7.1/cc -c -DNON_MATCHING -DNDEBUG -DOOT_DEBUG=0 -G 0 -non_shared -fullwarn -verbose -Xcpluscomm -Iinclude -Iinclude/libc -Isrc -Ibuild/gc-eu-mq -I. -Wab,-r4300_mul -woff 516,609,649,838,712 -mips2 -O2 -g3 {input} -o {output}
Assembler: mips-linux-gnu-as -march=vr4300 -mabi=32 {input} -o {output}
Preserving macros: gSPDisplayList, gSPSegment, gSPEndDisplayList, SQ, PRINTF, LOG, SET_FULLSCREEN_VIEWPORT, POLY_OPA_DISP, POLY_XLU_DISP, OVERLAY_DISP, OPEN_DISPS, CLOSE_DISPS, GRAPH_ALLOC. Use --preserve-macros='<regex>' to override.
/tmp/permuteru46fgu8v.c:
/tmp/oot_YZxuMAhC.c:

Done. Imported into nonmatchings/Play_Update
```

```
$ ../decomp-permuter/permuter.py nonmatchings/Play_Update/
Loading...
nonmatchings/Play_Update/base.c (Play_Update)
No perm macros found. Defaulting to randomization.
Will try 1 different base sources.
/tmp/permuterl7z0lk2a.c:
/tmp/oot_Ss8beWU8.c:
Traceback (most recent call last):
  File "/home/dragorn421/Documents/oot/../decomp-permuter/permuter.py", line 5, in <module>
    main()
  File "/home/dragorn421/Documents/decomp-permuter/src/main.py", line 771, in main
    run(options)
  File "/home/dragorn421/Documents/decomp-permuter/src/main.py", line 283, in run
    return run_inner(options, heartbeat)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dragorn421/Documents/decomp-permuter/src/main.py", line 360, in run_inner
    permuter = Permuter(
               ^^^^^^^^^
  File "/home/dragorn421/Documents/decomp-permuter/src/permuter.py", line 134, in __init__
    ) = self._create_and_score_base()
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dragorn421/Documents/decomp-permuter/src/permuter.py", line 157, in _create_and_score_base
    base_result = base_cand.score(self.scorer, o_file)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dragorn421/Documents/decomp-permuter/src/candidate.py", line 97, in score
    self.score_value, self.score_hash = scorer.score(o_file)
                                        ^^^^^^^^^^^^^^^^^^^^
  File "/home/dragorn421/Documents/decomp-permuter/src/scorer.py", line 46, in score
    objdump_output, cand_seq = self._objdump(cand_o)
                               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/dragorn421/Documents/decomp-permuter/src/scorer.py", line 39, in _objdump
    lines = objdump(o_file, self.arch, stack_differences=self.stack_differences)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dragorn421/Documents/decomp-permuter/src/objdump.py", line 443, in objdump
    output = subprocess.check_output([executable] + arch.arguments + [o_filename])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['mips-linux-gnu-objdump', '-drz', '-m', 'mips:4300', '/tmp/permuter4xrrvpby.o']' returned non-zero exit status 1.

$ ls /tmp/permuter4xrrvpby.o
ls: cannot access '/tmp/permuter4xrrvpby.o': No such file or directory
```

</details>

Reply:
actually no asm-processor also expects the last argument to be input so idk what's going on
https://github.com/simonlindholm/asm-processor/blob/fed1e3ddb9967bd427f4599b1b8ededa296bd9f9/build.py#L31

Reply:
Fixed by e9b52b493